common/fixation_entry_index.go

```diff
@@ -6,49 +6,42 @@ import (
 	"github.com/lavanet/lava/common/types"
 )
 
-/*
-The EntryIndex is a list which keeps all the latest version indices (e.g. indices without the version
-number suffix) of fixation entries of a particular module in the KVStore.
-Generally, modules can hold several types of objects that need to be fixated. To avoid a mix of indices between different
-kinds of fixation entries, the module that wishes to save an entry index must save it with a prefix that
-identifies it uniquely - this is the fixation key. For example, when the packages module wants to add a new
-entry index of a package object to the store, it acceses the store using the key prefix:
-"Entry_Index_package" (prefix = "package"). It saves the index under the key: "Entry_Index_package_<packageIndex>".
-*/
-
-// SetEntryIndex appends an entry index in the store with a new id and updates the count. It returns the index in the list of the added value (for example, if the first value of the list is added, it'll return 0 (the first index in the list))
-func (fs FixationStore) SetEntryIndex(ctx sdk.Context, index string) {
-	// get the index store with the fixation key
-	store := prefix.NewStore(ctx.KVStore(fs.GetStoreKey()), types.KeyPrefix(fs.createEntryIndexStoreKey()))
-
-	// convert the index value to a byte array
-	appendedValue := []byte(index)
-
-	// set the index in the index store
-	store.Set(types.KeyPrefix(types.EntryIndexKey+fs.prefix+index), appendedValue)
+// FixationStore manages lists of entries with versions in the store.
+// (See also documentation in common/fixation_entry.go)
+//
+// EntryIndex tracks the indices of every Entry stored in the FixationStore
+// (one index regardless of Entry version count). Its goal is to be able to
+// efficiently retrieve a list of all Entry indices that are stored. (Without
+// it, we would need to iterate through all the Entries *including* all their
+// versions - which may be wasteful).
+//
+// EntryIndex objects are stored with a distinct prefix to avoid confusion
+// when iterating over Entries. See example in `x/common/fixation_entry.go`.
+
+// setEntryIndex stores an Entry index in the store
+func (fs FixationStore) setEntryIndex(ctx sdk.Context, safeIndex string) {
+	store := prefix.NewStore(ctx.KVStore(fs.storeKey), types.KeyPrefix(fs.createEntryIndexStoreKey()))
```

Comment:
we can add another store utility function here for EntryIndex store and remove repeated calls to store create also here like you did in the prev file